src/topology/perfect.lean

```diff
@@ -0,0 +1,231 @@
+  acc_pt x (𝓟 (U ∩ C)) :=
+begin
+  have : 𝓝[≠] x ≤ 𝓟 U,
+  { rw le_principal_iff,
+    exact mem_nhds_within_of_mem_nhds hU, },
+  rw [acc_pt, ← inf_principal, ← inf_assoc, inf_of_le_left this],
+  exact h_acc,
+end
+
+/-- A set `C` is preperfect if all of its points are accumulation points of itself.
+If `C` is nonempty, this is equivalent to the closure of `C` being perfect.
+See `preperfect_iff_closure_perfect`.-/
+--Note : This is my own term, feel free to suggest a better one :P
+def preperfect (C : set α) : Prop := ∀ x ∈ C, acc_pt x (𝓟 C)
+
+/-- A set `C` is called perfect if it is closed and all of its
+points are accumulation points of itself.
+Note that we do not require `C` to be nonempty.-/
+structure perfect (C : set α) : Prop :=
+  (closed : is_closed C)
+  (acc : preperfect C)
+
+lemma preperfect_iff_nhds : preperfect C ↔ ∀ x ∈ C, ∀ U ∈ 𝓝 x, ∃ y ∈ U ∩ C, y ≠ x :=
+by simp only[preperfect, acc_pt_iff_nhds]
+
+/-- The intersection of a preperfect set and an open set is preperfect-/
+theorem preperfect.open_inter {U : set α} (hC : preperfect C) (hU : is_open U) :
+  preperfect (U ∩ C) :=
+begin
+  rintros x ⟨xU, xC⟩,
+  apply (hC _ xC).nhd_inter,
+  exact hU.mem_nhds xU,
+end
+
+/-- The closure of a preperfect set is perfect.
+For a converse, see `preperfect_iff_perfect_closure`-/
+theorem preperfect.perfect_closure (hC : preperfect C) : perfect (closure C) :=
+begin
+  split, { exact is_closed_closure },
+  intros x hx,
+  by_cases h : x ∈ C; apply acc_pt.mono _ (principal_mono.mpr subset_closure),
+  { exact hC _ h },
+  have : {x}ᶜ ∩ C = C := by simp[h],
+  rw [acc_pt, nhds_within, inf_assoc, inf_principal, this],
+  rw [closure_eq_cluster_pts] at hx,
+  exact hx,
+end
+
+/-- In a T1 space, being preperfect is equivalent to having perfect closure.-/
+theorem preperfect_iff_perfect_closure [t1_space α] :
+  preperfect C ↔ perfect (closure C) :=
+begin
+  split; intro h, { exact h.perfect_closure },
+  intros x xC,
+  have H := h.acc _ (subset_closure xC),
+  rw acc_pt_iff_frequently at *,
+  have : ∀ y , y ≠ x ∧ y ∈ closure C → ∃ᶠ z in 𝓝 y, z ≠ x ∧ z ∈ C,
+  { rintros y ⟨hyx, yC⟩,
+    simp only [← mem_compl_singleton_iff, @and_comm _ (_ ∈ C) , ← frequently_nhds_within_iff,
+      hyx.nhds_within_compl_singleton, ← mem_closure_iff_frequently],
+    exact yC, },
+  rw ← frequently_frequently_nhds,
+  exact H.mono this,
+end
+
+theorem perfect.closure_nhd_inter {U : set α} (hC : perfect C) (x : α) (xC : x ∈ C) (xU : x ∈ U)
+  (Uop : is_open U) : perfect (closure (U ∩ C)) ∧ (closure (U ∩ C)).nonempty :=
+begin
+  split,
+  { apply preperfect.perfect_closure,
+    exact (hC.acc).open_inter Uop, },
+  apply nonempty.mono subset_closure,
+  exact ⟨x, ⟨xU, xC⟩⟩,
+end
+
+/-- Given a perfect nonempty set in a T2.5 space, we can find two disjoint perfect subsets
+This is the main inductive step in the proof of the Cantor-Bendixson Theorem-/
+lemma perf_nonempty.splitting [t2_5_space α] (hC : perfect C) (hnonempty : C.nonempty) :
```

Comment:
Since `perf_nonempty` does not exist anymore, that name should be changed. To `perfect.splitting`?
```suggestion
lemma perfect.splitting [t2_5_space α] (hC : perfect C) (hnonempty : C.nonempty) :
```